lib/utils/RequestHandler.js

```diff
@@ -238,6 +238,14 @@ class RequestHandler {
             })
         })
     }
+
+    extendDefaultOptions (opts) {
+        const shouldExtend = Object.keys(opts).some((key) => opts[key] !== this.defaultOptions[key])
+
+        if (shouldExtend) {
+            this.defaultOptions = Object.assign(this.defaultOptions, opts)
```

Comment:
In fact `defaultOptions` are not *default* for the RequestHandler, they are just `options`. I think you should rename it